gen/cpp/sym/factors/internal_imu_factor.h

```diff
@@ -0,0 +1,2373 @@
+// -----------------------------------------------------------------------------
+// This file was autogenerated by symforce from template:
+//     function/FUNCTION.h.jinja
+// Do NOT modify by hand.
+// -----------------------------------------------------------------------------
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include <sym/pose3.h>
+#include <sym/rot3.h>
+
+namespace sym {
+
+/**
+ * An internal helper function to linearize the difference between the orientation, velocity, and
+ * position at one time step and those of another time step. The expected difference is calculated
+ * from the preintegrated IMU measurements between those time steps.
+ *
+ * NOTE: If you are looking for a residual for an IMU factor, do not use this. Instead use
+ * the one found in symforce/slam/imu_preintegration/imu_factor.h.
+ *
+ * Time step i is the time of the first IMU measurement of the interval.
+ * Time step j is the time after the last IMU measurement of the interval.
+ *
+ * Assumes sqrt_info is lower triangular and only reads the lower triangle.
+ *
+ * Args:
+ *     pose_i (sf.Pose3): Pose at time step i (maps body frame at time i to world frame)
+ *     vel_i (sf.V3): Velocity at time step i (world frame)
+ *     pose_j (sf.Pose3): Pose at time step j (maps body frame at time j to world frame)
+ *     vel_j (sf.V3): Velocity at time step j (world frame)
+ *     accel_bias_i (sf.V3): The accelerometer bias between timesteps i and j
+ *     gyro_bias_i (sf.V3): The gyroscope bias between timesteps i and j
+ *     DR (sf.Rot3): Preintegrated estimate for pose_i.R.inverse() * pose_j.R
+ *     Dv (sf.V3): Preintegrated estimate for vel_j - vel_i in the body frame at timestep i
+ *     Dp (sf.V3): Preintegrated estimate for pos change (before velocity and gravity corrections)
+ *         in the body frame at timestep i: R_i^T (p_j - p_i - v_i Delta t - 1/2 g Delta t^2),
+ *         where R_i = pose_i.R, p_i = pose_i.t, p_j = pose_j.t, and v_i = vel_i
+ *     sqrt_info (sf.M99): sqrt info matrix of DR('s tangent space), Dv, Dp
+ *     DR_D_gyro_bias (sf.M33): Derivative of DR w.r.t. gyro_bias evaluated at gyro_bias_hat
+ *     Dp_D_accel_bias (sf.M33): Derivative of Dp w.r.t. accel_bias evaluated at accel_bias_hat
+ *     Dp_D_gyro_bias (sf.M33): Derivative of Dp w.r.t. gyro_bias evaluated at gyro_bias_hat
+ *     Dv_D_accel_bias (sf.M33): Derivative of Dv w.r.t. accel_bias evaluated at accel_bias_hat
+ *     Dv_D_gyro_bias (sf.M33): Derivative of Dv w.r.t. gyro_bias evaluated at gyro_bias_hat
```

Comment:
Huh. I agree. Not exactly sure why I did it that way here. Done.

Reply:
It still seems like we have gyro then accel in some places and accel then gyro in others?  Not sure if something just hasn't been pushed yet

Reply:
Oh. I thought you were getting at how here its the `DR`, `Dp`, `Dv` instead of `DR`, `Dv`, `Dp`. Here gyro appears first as there's no `DR_D_accel_bias` (as it's just zero, as naturally the orientation is independent of the acceleration).

As for the ordering of `gyro` and `accel`, I only spotted 3 locations where gyro came first: 1) The ordering of the members in `PreintegratedImuMeasurements` (which doesn't really affect anything, but I just changed it so accel is first there for consistency), 2) the internal symbolic function `integrate_state` (which I have also just changed to be accel first), and 3) in the temporary variable `state_D_bias` in `imu_manifold_preintegration_update` (which I am keeping as is because the variable names don't line up nicely if I switch the order).

Reply:
Yep, (1) and (2) are what I was referring to.  Fine to leave (3), although as a super-nitpick we could pick gyro first instead to match the ordering there

Reply:
The interface of imu_manifold_preintegration_update is still backwards from the rest of the code, and I think we should fix it?  And then the body of ImuPreintegrator::IntegrateMeasurement which calls ImuManifoldPreintegrationUpdate

Reply:
Oops. Looks like I overlooked that function. Fixed.